editor/components/rich-text/index.js

```diff
@@ -829,6 +832,15 @@ export class RichText extends Component {
 		} ) );
 	}
 
+	addFootnote() {
+		this.editor.selection.collapse();
+		if ( this.editor.selection.getNode().tagName === 'SUP' ) {
+			return;
+		}
+		const uid = uuid();
+		this.editor.insertContent( `<sup data-wp-footnote-id="${ uid }"><a href="#${ uid }" class="wp-footnote"><span class="screen-reader-text">${ __( 'See footnote' ) }</span></a></sup> ` );
```

Comment:
Should this be `contenteditable=false`? Looks like complex markup that should be protected.

Reply:
Also here we need proper extension points in place. Maybe this could also make use of #6959 when merged.